src/Serilog/Parsing/MessageTemplateParser.cs

```diff
@@ -228,18 +228,7 @@ static bool TryGetDestructuringHint(char c, out Destructuring destructuring)
         }
     }
 
-    static bool IsValidInDestructuringHint(char c)
-    {
-        return c is '@' or '$';
-    }
-
-    static bool IsValidInFormat(char c)
-    {
-        return c != '}' &&
-               (char.IsLetterOrDigit(c) ||
-                char.IsPunctuation(c) ||
-                c is ' ' or '+');
-    }
+    static bool IsValidInFormat(char c) => c != (byte)'}';
```

Comment:
The byte conversion difference here is just noise; both versions end up emitting the same assembly (using godbolt.org for a quick check).

I'll drop out the cast and merge this so it's all ready for v4 :-) :+1:

Reply:
Wicked, on 500ish ns noise ends up so impactful :P